src/libraries/System.Linq.Expressions/src/System/Dynamic/Utils/TypeUtils.cs

```diff
@@ -822,43 +822,43 @@ public static Type FindGenericType(Type definition, Type type)
         /// op_False, because we have to do runtime lookup for those. It may
         /// not work right for unary operators in general.
         /// </summary>
-        public static MethodInfo GetBooleanOperator(Type type, string name)
+        public static MethodInfo? GetBooleanOperator(Type type, string name)
         {
             do
             {
-                MethodInfo result = type.GetAnyStaticMethodValidated(name, new[] { type });
+                MethodInfo? result = type.GetAnyStaticMethodValidated(name, new[] { type });
                 if (result != null && result.IsSpecialName && !result.ContainsGenericParameters)
                 {
                     return result;
                 }
 
-                type = type.BaseType;
+                type = type.BaseType!;
             } while (type != null);
 
             return null;
         }
 
-        public static Type GetNonRefType(this Type type) => type.IsByRef ? type.GetElementType() : type;
+        public static Type GetNonRefType(this Type type) => type.IsByRef ? type.GetElementType()! : type;
 
-        public static bool AreEquivalent(Type t1, Type t2) => t1 != null && t1.IsEquivalentTo(t2);
+        public static bool AreEquivalent(Type? t1, Type? t2) => t1 != null && t1.IsEquivalentTo(t2);
 
-        public static bool AreReferenceAssignable(Type dest, Type src)
+        public static bool AreReferenceAssignable(Type? dest, Type src)
```

Comment:
>Type? [](start = 50, length = 5)

Should be `Type` rather than `Type?` because the method calls `dest.IsValueType`.

Reply:
There is one case making it nullable:
 ```csharp
    public static bool IsValidInstanceType(MemberInfo member, Type instanceType)
    {
        Type? targetType = member.DeclaringType;
        if (AreReferenceAssignable(targetType, instanceType))
        {
            return true;
        }

        if (targetType == null)
        {
            return false;
        }
       …
```
Can I update this method to do null check first?

Reply:
I think it makes sense to change `IsValidInstanceType()` to check `if (targetType == null)` first.

---
In reply to: [362959568](https://github.com/dotnet/runtime/pull/1064#discussion_r362959568) [](ancestors = 362959568)